stac_model/examples.py

```diff
@@ -106,9 +105,10 @@ def eurosat_resnet():
         "SeaLake": 9,
     }
     class_objects = [
-        ClassObject(value=class_map[class_name], name=class_name)
+        MLMClassification.create(value=class_map[class_name], description=class_name, name=class_name)
```

Comment:
I think the use of Annotated is clear enough that we can stick with it. We could override the __init__, but I think the use of Annotated is more succinct.

Reply:
I have tried using the `Annotated` approach, but I ended up getting multiple `unhashable dict` errors when using `MLMClassification` within another `BaseModel` definition.

Reply:
I had to do the following to make it all work together: 
https://github.com/rbavery/dlm-extension/blob/2d6c70b63738e62908eef432ad046f186835ed58/stac_model/output.py#L34-L65

Reply:
We don't need to include MLMClassification in any other BaseModel definitions in this code base though I think? So it shouldn't be a problem? I haven't encountered that error when using Annotated when serializing or loading the data back in with pystac. The class above is more verbose and complex for contributors than the use of Annotated imo. Can we keep it as is since it works?

Reply:
Yes. It must be included in the model output. We employ the same definition as `classification:classes`, but they are not currently validated by that extension, because field would be nested under a specific MLM output rather than at the Item properties or Asset level (see https://github.com/stac-extensions/classification/issues/48). 

Using the above definition, serialize/load works both ways, as evaluated by this test:
https://github.com/rbavery/dlm-extension/blob/4eb30dab98617dfee4cc6b1e5dbc1d50d4770779/tests/test_schema.py#L36-L39
using this definition:
https://github.com/rbavery/dlm-extension/blob/4eb30dab98617dfee4cc6b1e5dbc1d50d4770779/stac_model/examples.py#L106-L116

Reply:
Ok gotcha. I'm still a bit confused why the use of `Annotated` wasn't working out but trust this is best.

Since you have this working in your PR I'll close this @fmigneault